tfhe/src/integer/server_key/comparator.rs

```diff
@@ -1518,13 +2099,13 @@ mod tests {
                 create_parametrized_test!([<unchecked_ $comparison_name _256_bits>]
                 {
                     PARAM_MESSAGE_2_CARRY_2,
-                    // PARAM_MESSAGE_3_CARRY_3,
+                    PARAM_MESSAGE_3_CARRY_3,
                     PARAM_MESSAGE_4_CARRY_4
                 });
                 create_parametrized_test!([<unchecked_ $comparison_name _parallelized_256_bits>]
                 {
                     PARAM_MESSAGE_2_CARRY_2,
-                    // PARAM_MESSAGE_3_CARRY_3,
+                    PARAM_MESSAGE_3_CARRY_3,
```

Comment:
what changed to be 3_3 friendly ?

Reply:
Comparisons always worked for 3_3, be it in unchecked, smart or default, it always worked on 3_3.

The problem in our tests, is that for the default and smart case we generate dirty inputs, so inputs for which
there are carries and the value could have overflowed.

The problem with 3_3 in that case, is that we are not truly encrypting 256 bits but 3_3 * 86 = 258 bits.
This means that when a u256 would have overflowed (in our case bits of carry are set in the very last block)
the u258 will not have overflowed, its has 2 more bits compared to the 256bits.

So then in the test we are comparing the result of comparsisons between overflowed u256 values (in clear)
with the result of the comparisons with u258 values.

Its like comparing bananas to apple, its not the same things so the results won't match

Reply:
I understand the issue you describe, but here the test is re-enabled just wanted to understand what changed ?

Reply:
Nothing changed in the implementation, there was no need to disable the unchecked 3_3 tests so I re enabled them so that they run and we can make sure we dont break comparisons on 3_3

Reply:
ah understood, maybe I was the one who just disabled everything without thinking when we got some flakies around that part of the code, thanks !